module/zfs/vdev_raidz.c

```diff
@@ -2745,6 +2846,156 @@ vdev_raidz_worst_error(raidz_row_t *rr)
 	return (error);
 }
 
+/*
+ * Find the median value from a set of n values
+ */
+static uint64_t
+latency_median_value(const uint64_t *data, size_t n)
+{
+	uint64_t m;
+
+	if (n % 2 == 0)
+		m = (data[(n>>1) - 1] + data[n>>1]) >> 1;
+	else
+		m = data[((n + 1) >> 1) - 1];
+
+	return (m);
+}
+
+/*
+ * Calculate the outlier fence from a set of n latency values
+ *
+ * fence = Q3 + 2 x (Q3 - Q1)
+ */
+static uint64_t
+latency_quartiles_fence(const uint64_t *data, size_t n)
+{
+	uint64_t q1 = latency_median_value(&data[0], n >> 1);
+	uint64_t q3 = latency_median_value(&data[(n + 1) >> 1], n >> 1);
+
+	/*
+	 * To avoid detecting false positive outliers when N is small and
+	 * and the latencies values are very close, make sure the fence
+	 * is at least 25% larger than Q1.
+	 */
+	uint64_t iqr = MAX(q3 - q1, q1 >> 3);
+
+	return (q3 + (iqr << 1));
+}
+
+#define	LAT_SAMPLES_STACK	64
+#define	LAT_SAMPLES_MIN		5
+#define	LAT_OUTLIER_LIMIT	50
+
+static int
+latency_compare(const void *arg1, const void *arg2)
+{
+	const uint64_t *l1 = (uint64_t *)arg1;
+	const uint64_t *l2 = (uint64_t *)arg2;
+
+	return (TREE_CMP(*l1, *l2));
+}
+
+/*
+ * Check for any latency outlier from latest set of child reads.
+ *
+ * Uses a Tukey's fence, with K = 2, for detecting extreme outliers. This
```

Comment:
@don-brady was a K=2 experimentally determined to be a good value for the fence?  K=3 is usually used to characterize extreme outliers, but perhaps that didn't catch enough?  It'd be nice to have some more visibility in to this so we could collect some usual statistics from real systems with problem drives.